app/src/main/java/com/framgia/englishconversation/data/source/remote/comment/CommentRemoteDataSource.java

```diff
@@ -109,28 +107,25 @@ public void onCancelled(DatabaseError databaseError) {
         });
     }
 
-    public Observable<Comment> registerModifyTimelines(final Comment lastComment) {
-        return Observable.create(new ObservableOnSubscribe<Comment>() {
+    public Observable<HashMap<Integer, Comment>> registerModifyTimelines(
+            final Comment lastComment) {
+        return Observable.create(new ObservableOnSubscribe<HashMap<Integer, Comment>>() {
             @Override
-            public void subscribe(final ObservableEmitter<Comment> e) throws Exception {
+            public void subscribe(final ObservableEmitter<HashMap<Integer, Comment>> e)
+                    throws Exception {
                 final Query query = mReference.orderByChild(Constant.DatabaseTree.CREATED_AT)
                         .endAt(lastComment != null ? -lastComment.getCreatedAt()
                                 : -Calendar.getInstance().getTimeInMillis());
                 mUpdateComment = new ChildEventListener() {
                     @Override
                     public void onChildAdded(DataSnapshot dataSnapshot, String s) {
-                        if (lastComment != null && lastComment.getId()
-                                .equals(dataSnapshot.getKey())) {
-                            return;
-                        }
                         Comment comment = dataSnapshot.getValue(Comment.class);
-                        if (comment.getStatusModel() == null
-                                || comment.getStatusModel().getStatus() == Status.NORMAL) {
-                            comment.setCreatedAt(
-                                    Utils.generateOppositeNumber(comment.getCreatedAt()));
-                            comment.setId(dataSnapshot.getKey());
-                            e.onNext(comment);
-                        }
+                        comment.setCreatedAt(
+                                Utils.generateOppositeNumber(comment.getCreatedAt()));
+                        comment.setId(dataSnapshot.getKey());
+                        HashMap<Integer, Comment> commentHashMap = new HashMap<>();
```

Comment:
WARNING: [UseSparseArrays] Use `new SparseArray<Comment>(...)` instead for better performance
 For maps where the keys are of type integer, it's typically more efficient to use the Android `SparseArray` API. This check identifies scenarios where you might want to consider using `SparseArray` instead of `HashMap` for better performance.

This is **particularly** useful when the value types are primitives like ints, where you can use `SparseIntArray` and avoid auto-boxing the values from `int` to `Integer`.

If you need to construct a `HashMap` because you need to call an API outside of your control which requires a `Map`, you can suppress this warning using for example the `@SuppressLint` annotation.